scripts/hive_adi_test.py

```diff
@@ -77,7 +79,7 @@ def main():
 
     def remove_control_characters(s):
         if isinstance(s, str):
-            s = unicode(s, 'utf-8', errors='replace')
+            s = str(s, 'utf-8', errors='replace')
```

Comment:
It looks like some of your changes didn't make it into commit. I recall this was:
```python
s = s.decode('utf-8', errors='replace')
```

Reply:
Previously, this if-block would check to see if the value was a `str` and convert it to a `unicode`. Now it's kind of silly in that it checks to see if the value is a `str` and if so, converts it to a `str`.

So, what is the big picture here? What is this code doing? What's the context? Can `s` have a binary non-character value? If we ran this code in Python 3, could it be `bytes`? Or is it the case that in Python 2, it was sometimes a string and sometimes a unicode--but always character data?

This function is removing control characters. Why would the value have control characters in it? How curious. The next line converts the result back to a `str`. Even more curious.

I think what we want here is to convert it to a unicode if and only if we're in Python 2 and the value is a str. If we're in Python 3 or if the value is unicode, then we can leave it as is--we don't need to convert it.

So maybe something like this works?:

```
if six.PY2 and isinstance(s, str):
    s = six.text_type(s, 'utf-8', errors='replace')
```

Taking a bigger step back, we don't actually use this file at all and there are no tests for it--it's effectively half-dead code that's seen better days and is waiting for @lonnen to give us the thumbs-up to remove all the ADI stuff.

Thus, nothing we do to this file has any impact on the world so we can leave it as is or make the change you suggested or do a different change--it's all fine.

Reply:
Oops--we can't leave it as is since we need to get linting to pass, so the options are:

1. keep the change you have
2. make a different change